test/unit/transform/ferc1_test.py

```diff
@@ -1231,3 +1235,92 @@ def test_unexpected_total_components():
         )
     )
     assert unexpected_total_components(no_extra_components, dimensions).empty
+
+
+def test_filter_for_freshest_data_xbrl_simple():
+    df = pd.DataFrame.from_records(
+        [
+            {
+                "entity_id": "C000001",
+                "utility_type_axis": "electric",
+                "filing_name": "Utility_Co_0001",
+                "date": datetime.date(2021, 12, 31),
+                "publication_time": datetime.datetime(2022, 2, 1, 0, 0, 0),
+                "str_factoid": "original 2021 EOY value",
+            },
+            {
+                "entity_id": "C000001",
+                "utility_type_axis": "electric",
+                "filing_name": "Utility_Co_0002",
+                "date": datetime.date(2021, 12, 31),
+                "publication_time": datetime.datetime(2022, 2, 1, 1, 1, 1),
+                "str_factoid": "updated 2021 EOY value",
+            },
+        ]
+    )
+    observed_table = filter_for_freshest_data_xbrl(
+        df,
+        primary_keys=[
+            "entity_id",
+            "filing_name",
+            "publication_time",
+            "date",
+            "utility_type_axis",
+        ],
+    )
+
+    assert len(observed_table) == 1
+    assert observed_table.str_factoid.to_numpy().item() == "updated 2021 EOY value"
+
+
+example_schema = pandera.DataFrameSchema(
+    {
+        "entity_id": pandera.Column(
+            str, pandera.Check.isin("C0123456789"), nullable=False
+        ),
+        "date": pandera.Column("datetime64[ns]", nullable=False),
+        "utility_type": pandera.Column(
+            str,
+            pandera.Check.isin(["electric", "gas", "total", "other"]),
+            nullable=False,
+        ),
+        "publication_time": pandera.Column("datetime64[ns]", nullable=False),
+        "int_factoid": pandera.Column(int),
+        "float_factoid": pandera.Column(float),
+        "str_factoid": pandera.Column(str),
+    }
+)
+
+
+# ridiculous deadline - dataframe generation is always slow and sometimes
+# *very* slow
+@pytest.mark.slow
+@hypothesis.settings(print_blob=True, deadline=2_000)
+@hypothesis.given(example_schema.strategy(size=3))
+def test_filter_for_freshest_data_xbrl(df):
+    # XBRL context is the identifying metadata for reported values
+    xbrl_context_cols = ["entity_id", "date", "utility_type"]
+    filing_metadata_cols = ["publication_time", "filing_name"]
+    primary_key = xbrl_context_cols + filing_metadata_cols
+    deduped = filter_for_freshest_data_xbrl(df, primary_keys=primary_key)
+    example_schema.validate(deduped)
```

Comment:
hm @jdangerx  this is failing rn bc the `validate` expected the `deduped` to have `publication_time` in there.. which i am now dropping as a part of `filter_for_freshest_data_xbrl`. I could just not drop it in there are always remember to drop it after `filter_for_freshest_data_xbrl` is called. but idk it does seem like it should be dropped always so it'd be nice to just make the test work. any thoughts/ideas?

```python
test/unit/transform/ferc1_test.py:1306: in test_filter_for_freshest_data_xbrl
    example_schema.validate(deduped)
../../miniforge3/envs/pudl-dev/lib/python3.12/site-packages/pandera/api/pandas/container.py:126: in validate
    return self._validate(
../../miniforge3/envs/pudl-dev/lib/python3.12/site-packages/pandera/api/pandas/container.py:155: in _validate
    return self.get_backend(check_obj).validate(
../../miniforge3/envs/pudl-dev/lib/python3.12/site-packages/pandera/backends/pandas/container.py:105: in validate
    error_handler = self.run_checks_and_handle_errors(
../../miniforge3/envs/pudl-dev/lib/python3.12/site-packages/pandera/backends/pandas/container.py:180: in run_checks_and_handle_errors
    error_handler.collect_error(
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <pandera.api.base.error_handler.ErrorHandler object at 0x28f24cc50>, error_type = <ValidationScope.SCHEMA: 'schema'>
reason_code = <SchemaErrorReason.COLUMN_NOT_IN_DATAFRAME: 'column_not_in_dataframe'>
schema_error = SchemaError("column 'publication_time' not in dataframe. Columns in dataframe: ['entity_id', 'date', 'utility_type', 'int_factoid', 'float_factoid', 'str_factoid']")
original_exc = None

    def collect_error(
        self,
        error_type: ErrorCategory,
        reason_code: SchemaErrorReason,
        schema_error: SchemaError,
        original_exc: Union[BaseException, None] = None,
    ):
        """Collect schema error, raising exception if lazy is False.

        :param error_type: type of error
        :param reason_code: string representing reason for error
        :param schema_error: ``SchemaError`` object.
        """
        if not self._lazy:
>           raise schema_error from original_exc
E           pandera.errors.SchemaError: column 'publication_time' not in dataframe. Columns in dataframe: ['entity_id', 'date', 'utility_type', 'int_factoid', 'float_factoid', 'str_factoid']
E           Falsifying example: test_filter_for_freshest_data_xbrl(
E               df=
E                     entity_id       date utility_type publication_time  int_factoid  float_factoid str_factoid
E                   0         C 1970-01-01     electric       1970-01-01            0            0.0
E                   1         C 1970-01-01     electric       1970-01-01            0            0.0
E                   2         C 1970-01-01     electric       1970-01-01            0            0.0
E               ,  # or any other generated value
E           )
E
E           You can reproduce this example by temporarily adding @reproduce_failure('6.112.1', b'AXicY2AYTAAAAI4AAQ==') as a decorator on your test case

../../miniforge3/envs/pudl-dev/lib/python3.12/site-packages/pandera/api/base/error_handler.py:54: SchemaError
```

Reply:
okay this is simple. does this mean that example_schama is a dataframe when it is in here (i've never worked with hypothesis)